### ISSUE TITLE
Bug: CodeIgniter4 shows errors when MariaDB/MySQL has 'ANSI_QUOTES' enabled in sql_mode.

### ISSUE THREAD

**Issue by mv-steven** [codeigniter4/CodeIgniter4#5423]
### PHP Version

7.4

### CodeIgniter4 Version

4.1.5

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

cli-server (PHP built-in webserver)

### Database

MariaDB 10.4.22

### What happened?

My company's database systems have the `ANSI_QUOTES` option set in `sql_mode` which treats double-quotes like backticks for quoting identifier names.  I'd like to change this in our systems, but we have legacy code that requires this.

When accessing any page in CodeIgniter that connects to the DB, an exception is thrown with the message `Main connection [MySQLi]: Unknown column 'STRICT_ALL_TABLES,' in 'field list'`.  This is coming from [Connection.php:99](https://github.com/codeigniter4/CodeIgniter4/blob/d7ad2286eef3040a8c57b8231168a972b6ca19d1/system/Database/MySQLi/Connection.php#L99).  If I remove `ANSI_QUOTES` from `sql_mode`, this fixes this, but this presents a problem for our legacy software.  If I just flip the quotes - quote the entire string with double-quotes and use single-quotes for the DB params, this seems to fix the problem.  I also notice that in line [181](https://github.com/codeigniter4/CodeIgniter4/blob/d7ad2286eef3040a8c57b8231168a972b6ca19d1/system/Database/MySQLi/Connection.php#L181), single quotes are used for 'ssl_cipher'.

### Steps to Reproduce

In the database, run `SET GLOBAL sql_mode=CONCAT(@@sql_mode,'ANSI_QUOTES');`, then load any page that requires a DB connection, it should throw the error above because it's treating anything in double-quotes as an identifier.

### Expected Output

The application should run without an error.

### Anything else?

I think if the quotes are flipped for lines 103 and 108, that will fix the issue for any DB systems with ANSI_QUOTES turned on.  It is also backwards-compatible with any DB system that doesn't have ANSI_QUOTES turned on.

I did a very cursory scan of anywhere else that double-quotes are being used in mysqli driver interactions, but I didn't see any; I'm by no means an expert in how CodeIgniter runs queries against MySQL/MariaDB, so someone else with more experience in that code may have a better idea of what else may be affected.

**Comment by kenjis** [comment]
I confirmed this bug.

```sql
SET GLOBAL sql_mode=CONCAT(@@sql_mode,',ANSI_QUOTES');
```

```
CodeIgniter\Database\Exceptions\DatabaseException #8

Unable to connect to the database.
Main connection [MySQLi]: Unknown column 'STRICT_ALL_TABLES,' in 'field list' search →
```

**Comment by kenjis** [comment]
@mv-steven Thank you for reporting.
I created a PR to fix this: #5425

**Comment by kenjis** [comment]
@mv-steven This bug was fixed.
It would be included in the next relase v4.1.6.

**Comment by mv-steven** [comment]
Awesome. Thank you!
